src/WebJobs.Script.WebHost/FunctionsSyncService.cs

```diff
@@ -42,8 +42,10 @@ internal bool ShouldSyncTriggers
         {
             get
             {
-                return _primaryHostStateProvider.IsPrimary &&
+                var shouldSyncTrigger = _primaryHostStateProvider.IsPrimary &&
                        (_scriptHostManager.State == ScriptHostState.Running);
+                _logger.LogDebug("ShouldSyncTriggers: '{shouldSyncTrigger}',PrimaryHostStateProvider.IsPrimary: '{primaryHostStateProvider.IsPrimary}',ScriptHostState: '{scriptHostManager.State}'", shouldSyncTrigger, _primaryHostStateProvider.IsPrimary, _scriptHostManager.State);
```

Comment:
I don't like that we're writing a log each time this property is accessed. For example, currently it's accessed in a timer callback. Perhaps now that's only invoked once, but in the future this property may be used in other ways, and we don't want a log entry each time it's called.

Reply:
As of now, the caller of this prop is also in the same class. So all the information is also available for the caller method (`OnSyncTimerTick`). The logging can be moved from property get to the caller(`OnSyncTimerTick`).